src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -433,5 +433,19 @@ private static unsafe bool TryGetUserNameFromPasswd(byte* buf, int bufLen, out s
         }
 
         public static string UserDomainName => MachineName;
+
+        /// <summary>Invoke <see cref="Interop.Sys.SysConf"/>, throwing if it fails.</summary>
+        private static int CheckedSysConf(Interop.Sys.SysConfName name)
+        {
+            long result = Interop.Sys.SysConf(name);
+            if (result == -1)
+            {
+                Interop.ErrorInfo errno = Interop.Sys.GetLastErrorInfo();
+                throw errno.Error == Interop.Error.EINVAL ?
+                    new ArgumentOutOfRangeException(nameof(name), name, errno.GetErrorMessage()) :
+                    Interop.GetIOException(errno);
```

Comment:
maybe GetIOException should throw ArgumentException for EINVAL.

Reply:
GetIOException returns IOException :) It'd be strange for it to return other non-derived types.